### PR TITLE
Change get_range signature to accept comparison closures.

### PR DESCRIPTION
**Pull request by ysono**
Doing this in preparation for secondary index. There, we have to compare our boundary definitions against a sub-portion of `K` in `LSMTree`, and not the whole `K`.

The signature is more complicated, but that's more because of the lifetime of the boundary references (the reference in `Option<&PrimaryKey>`) and how it's getting passed to the closure. We are creating one closure object for one boundary reference, and we're not copying the closure or the boundary reference.

### CODE REVIEW

File: src/storage/sstable.rs [hunk 1]
```diff
@@ -231,7 +241,13 @@ where
         let mut sparse_index = SparseIndex::new();
         let mut offset = 0 as FileOffset;
 
-        for (kv_i, res_kv) in Self::merge_range(tables, &None, &None)?.enumerate() {
+        for (kv_i, res_kv) in Self::merge_range(
+            tables,
+            None::<&Box<dyn Fn(&K) -> Ordering>>,
+            None::<&Box<dyn Fn(&K) -> Ordering>>,
```
Comment (ysono) [inline src/storage/sstable.rs:247]: There is an unstable feature that will eventually allow us to shorten this to `None::<!>`.

**Review by btc — APPROVED** [review]
